app/models/concerns/pdf_representable.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module PdfRepresentable
+  extend ActiveSupport::Concern
+
+  def generate_pdf
+    raise "No authoritative_json to create PDF for #{oid}" unless authoritative_json
+    # temp_json_file = Tempfile.new("#{oid}_pdf_json")
+    # temp_json_file.write(pdf_generator_json)
+    # temp_json_file.close
+    # temp_pdf_file = "#{temp_json_file.path}.pdf"
+    # cmd = "java -jar jpegs2pdf-1.0.jar #{temp_json_file.path} #{temp_pdf_file}"
+    # success = system(cmd)
+    # temp_json_file.delete
+    # if success
+    #   raise "Java app did not create PDF file for #{oid}" unless File.exist? temp_pdf_file
+    #   metadata = { 'generated': generated }
+    #   S3Service.upload_image(temp_pdf_file.to_s, remote_pdf_path, "application/pdf", metadata)
+    #   File.delete temp_pdf_file
+    # else
+    #   File.delete temp_pdf_file if File.exist?(temp_pdf_file)
+    #   raise "PDF Java app returned non zero response code for #{oid}"
+    # end
+  end
+
+  def remote_pdf_path
+    "pdfs/#{Partridge::Pairtree.oid_to_pairtree(oid)}/#{oid}.pdf"
+  end
+
+  def pdf_generator_json
+    generated = Time.now.utc.to_s
+    title = extract_flat_field_value(authoritative_json, "title", "No Title")
+    properties = pdf_properties title, generated
+    children = child_pages
+    return nil if children.empty?
```

Comment:
consider returning empty hash instead of nil, caller can do the x.is_empty? if they feel like it

Reply:
Changed it to return json with emtpy child hash even when there are no children.

Reply:
Updated jar to generate PDF with cover page if there are no children.